### PR TITLE
BXMSDOC-8107 OptaPlanner Core Micrometer Metrics - BAPL-1986

### PR DESCRIPTION
**Pull request by emmurphy1**

### CODE REVIEW

File: assemblies/assembly-planner-configuration.adoc [hunk 1]
```diff
@@ -542,6 +542,59 @@ most appenders, including the console, cause congestion with `debug` and `trace`
...
+----
+The following example shows how to add the Prometheus registry:
++
+[source,java,nowrap]
+----
+PrometheusMeterRegistry prometheusRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+try {
+    HttpServer server = HttpServer.create(new InetSocketAddress(8080), 0);
+    server.createContext("/prometheus", httpExchange -> {
+        String response = prometheusRegistry.scrape(); (1)
```
Comment (dupliaka) [inline assemblies/assembly-planner-configuration.adoc]: remove (1)

**Review by briandooley — CHANGES_REQUESTED** [review]

File: assemblies/assembly-planner-configuration.adoc [hunk 2]
```diff
@@ -542,6 +542,145 @@ most appenders, including the console, cause congestion with `debug` and `trace`
...
+OptaPlanner exposes metrics through https://micrometer.io/[Micrometer], a metrics instrumentation library for Java applications. You can use Micrometer with popular monitoring systems to monitor the OptaPlanner solver.
+
+==== Configuring a Quarkus OptaPlanner application for Micrometer
+To configure your OptaPlanner Quarkus application to use Micrometer and a specified monitoring system, add the Micrometer dependency to the `pom.xml` file.
+
+.Prerequisites
+* You have a Quarkus OptaPlanner application.
+
+.Procedure
+. Add the following dependency to your application's `pom.xml` file where `<MONITORING_SYSTEM>` is a monitoring system supported by Micrometer:
```
Comment (Christopher-Chianelli) [inline assemblies/assembly-planner-configuration.adoc]: Need to be supported by Quarkus AND Micrometer. Prometheus is the only registry supported by Quarkus currently.

File: assemblies/assembly-planner-configuration.adoc [hunk 3]
```diff
@@ -542,6 +542,145 @@ most appenders, including the console, cause congestion with `debug` and `trace`
...
+<dependency>
+ <groupId>io.micrometer</groupId>
+ <artifactId>micrometer-registry-<MONITORING_SYSTEM></artifactId>
+</dependency>
+----
+. Add the following line to the application's `application.properties` file:
++
+[source]
+----
+management.endpoints.web.exposure.include=metrics,<MONITORING_SYSTEM>
```
Comment (Christopher-Chianelli) [inline assemblies/assembly-planner-configuration.adoc]: The <MONITORING_SYSTEM> here and below are specific to Prometheus; see https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html#actuator.metrics.export for details on how to set up the various registries. The other registries go to `/actuator/metrics`

**Review by Christopher-Chianelli — CHANGES_REQUESTED** [review]
Note that most of these instructions are specific to Prometheus (Prometheus is the only registry officially supported by Quarkus; the other registries are from `io.quarkiverse.micrometer.registry` (https://search.maven.org/search?q=io.quarkiverse.micrometer.registry)).

**Comment by Christopher-Chianelli** [comment]
As for why Prometheus is special, it the only registry that pull data from the application; the other registries expect the application to push data to it (thus they don't need an endpoint, but the application must know their endpoint).

File: assemblies/assembly-planner-configuration.adoc [hunk 4]
```diff
@@ -596,12 +598,8 @@ To configure your Spring Boot OptaPlanner  application to use Micrometer and a s
  <artifactId>micrometer-registry-<MONITORING_SYSTEM></artifactId>
 </dependency>
 ----
-. Add the following line to the application's `application.properties` file:
-+
-[source]
-----
-management.endpoints.web.exposure.include=metrics,<MONITORING_SYSTEM>
-----
+. Add configuration information to the application's `application.properties` file. For information, see the https://micrometer.io[Micrometer] web site.
```
Comment (Christopher-Chianelli) [inline assemblies/assembly-planner-configuration.adoc:601]: For Spring Boot, the website is https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html#actuator.metrics.export

File: assemblies/assembly-planner-configuration.adoc [hunk 5]
```diff
@@ -554,7 +554,9 @@ To configure your OptaPlanner Quarkus application to use Micrometer and a specif
 * You have a Quarkus OptaPlanner application.
 
 .Procedure
-. Add the following dependency to your application's `pom.xml` file where `<MONITORING_SYSTEM>` is a monitoring system supported by Micrometer:
+. Add the following dependency to your application's `pom.xml` file where `<MONITORING_SYSTEM>` is a monitoring system supported by Micrometer and Quarkus:
++
+NOTE: Prometheus is currently the only monitoring system supported by Quarkus.
```
Comment (Christopher-Chianelli) [inline assemblies/assembly-planner-configuration.adoc:559]: Officially supported; the rest of the monitoring systems are community supported in the Quarkiverse; the rest of the registries either need to be manually set up (like plain Java) or use a community extension (https://github.com/quarkiverse/quarkus-micrometer-registry)

**Review by Christopher-Chianelli — CHANGES_REQUESTED** [review]
Spring Boot has specific docs for micrometer configuration;

**Review by dupliaka — APPROVED** [review]

**Pull request merged**